src/render.js

```diff
@@ -53,14 +53,13 @@ module.exports = function render() {
     var changed = [];
 
     this.featureIds.forEach((id) => {
-      let featureInternal = this.features[id].internal(mode);
+      let feature = this.features[id];
+      let featureInternal = feature.internal(mode);
       var coords = JSON.stringify(featureInternal.geometry.coordinates);
 
-      if (this.ctx.store.needsUpdate(featureInternal)) {
+      if (feature.isValid() && this.ctx.store.needsUpdate(feature.toGeoJSON())) {
```

Comment:
I'd like to use `featureInternal` here so we don't `JSON.parse(JSON.stringify())` more than needed.